sklearn/impute/_base.py

```diff
@@ -218,13 +221,21 @@ class SimpleImputer(_BaseImputer):
 
     See Also
     --------
-    IterativeImputer : Multivariate imputation of missing values.
+    IterativeImputer : Multivariate imputer that estimates each feature
+        from all the others.
+    KNNImputer : Multivariate imputer that estimates missing features using
+        nearest samples.
 
     Notes
     -----
     Columns which only contained missing values at :meth:`fit` are discarded
     upon :meth:`transform` if strategy is not `"constant"`.
 
+    In a prediction context, simple imputation usually performs poorly when
+    associated with a weak learner. However, with a powerful learner, it can
```

Comment:
It might be good to give additional details regarding what we call a weak learner even if it is only in brackets.

Maybe a proposal that can be modified:

```
a weak learner, i.e. generally refers to a constrained learner that easily underfit (e.g. linear models),
```

Reply:
The same could be done with `powerful learner` referring to GBDT.